components/ProfileDropdownMenu.tsx

```diff
@@ -0,0 +1,138 @@
+import React from 'react'
+import Dropdown from 'react-bootstrap/Dropdown'
+import UserInfoImage from './UserInfoImage'
+import { ChevronDownIcon } from '@primer/octicons-react'
+import LogoutContainer from './LogoutContainer'
+import { Button } from './theme/Button'
+import { UserInfo } from '../@types/user'
+import { useUserInfoQuery } from '../graphql/index'
+import { useRouter } from 'next/router'
+import _ from 'lodash'
+import styles from '../scss/profileDropDown.module.scss'
+
+type ProfileDropDownMenuProps = {
+  username: string
+  isAdmin: boolean
+}
+
+type CustomToggleProps = {
+  children?: React.ReactNode
+  className?: string | undefined
+  onClick?: (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => {}
+}
+
+const ProfileDropdownMenu: React.FC<ProfileDropDownMenuProps> = ({
+  username,
+  isAdmin
+}) => {
+  const { data } = useUserInfoQuery({
+    variables: { username },
+    skip: !username
+  })
+  const router = useRouter()
+  const location = router.asPath
+
+  const fullname = _.get(data, 'userInfo.user.name', '')
+
+  const userInfo: UserInfo = {
+    // 'A' stands for Anonymous, in case user did not put in full name
+    username,
+    firstName: fullname.split(' ')[0] || 'A',
+    lastName: fullname.split(' ')[1] || ' ',
+    discordUserId: data?.userInfo?.user?.discordUserId ?? '',
+    discordUsername: data?.userInfo?.user?.discordUsername ?? '',
+    discordAvatarUrl: data?.userInfo?.user?.discordAvatarUrl ?? ''
+  }
+
+  const dropdownAdminMenuItems = [
+    { title: 'Lessons', path: '/admin/lessons' },
+    { title: 'Users', path: '/admin/users' },
+    { title: 'Alerts', path: '/admin/alerts' }
+  ]
+
+  const isActive = (path: string) => {
+    if (path === location) return `${styles['active']}`
+    return ''
+  }
+
+  const adminDropdownMenu = dropdownAdminMenuItems.map(({ title, path }) => (
+    <Dropdown.Item
+      className={`${styles['dropdown-item']} nav-link `}
+      href={path}
+      key={title}
+      bsPrefix={isActive(path)}
+    >
+      {title}
+    </Dropdown.Item>
+  ))
+
+  const buttonMenuToggle = React.forwardRef(
+    (props: CustomToggleProps, ref: React.Ref<HTMLDivElement>) => (
```

Comment:
Your CustomToggleProps have fields that are already in `React.FC`. Is the typescript definition for `CustomToggleProps` needed?

Reply:
Sorry, i'm not sure what you mean, the `CustomToggleProps` already uses react typescript definitions except for the classname.

Reply:
If you look at the fields in your `customToggleProps`, you'll notice the fields are already attributes of a normal div component, except for the `children`. So there should already be some existing typescript definition out there you can use for this use case. For example an existing typescript definition you used is `React.Ref<HTMLDivElement>` to define the type for ref. There should similarly be one available for div element props

Reply:
Thanks for explanation, i got it fixed.